### ISSUE TITLE
SchemaSpy 6.1: wrong name for PK index

### ISSUE THREAD

**Issue by mapausch** [schemaspy/schemaspy#546]
Today I wanted to try SchemaSpy 6.1 SNAPSHOT from https://github.com/schemaspy/schemaspy/raw/gh-pages/schemaspy-6.1.0-SNAPSHOT.jar (schemaspy@latest)

During "Gathering schema details" I see some warnings like
_WARN  - Found PK for table 'GIWSDURATIONHISTORY' with index name 'PK_GIWSDURATIONHISTORYL', but index hasn't been found_
for different PK indexes.

Had a look on the table GIWSDURATIONHISTORY.
![grafik](https://user-images.githubusercontent.com/18686836/55808384-40efa000-5ae4-11e9-9ea5-bbc4ceb0529b.png)

For this table the name of the PK **Constraint** is PK_GIWSDURATIONHISTORY**L**, but the name of the index that supports the constraint is PK_GIWSDURATIONHISTORY (without L at the end)

So for me it seems that SchemaSpy 6.1 reads the name of the PK Constraint instead of name of the index that supports the PK constraint.

With SchemaSpy 6.0-rc1-something (pagination performance) from https://github.com/schemaspy/schemaspy/pull/394
I don't get the mentioned warnings when I use it on the same schema.
Also with other versions of SchemaSpy I couldn't remember that I ever seen this warnings.

So I assume that SchemaSpy 6.1 reads other information from data dictionary as SchemaSpy 6.0 did. 

## Steps to Reproduce (for bugs)
1. create table with PK constraint and index name that differs from the name of the constraint (see DDL below)
2. let schemaspy run :-)

## Context
* Database type: Oracle 12.1
* Database driver version (is you use a special one): Oracle JDBC 12.1 (with Java 1.8.0_144)
* Your DDL (sample how to reproduce the issue):
```
CREATE TABLE "GIWSDURATIONHISTORY" 
   (	"GIWSDURATIONHISTORYID" NUMBER(12,0) NOT NULL ENABLE, 
	"DURATIONHISTORYID" NUMBER(12,0) NOT NULL ENABLE, 
	"WORKSTATIONID" NUMBER(12,0), 
	"USERNAME" VARCHAR2(30 CHAR) NOT NULL ENABLE, 
	"GOODSINPOSID" NUMBER(12,0), 
	"QTY" NUMBER(15,3) NOT NULL ENABLE, 
	"RESERVED" VARCHAR2(255 CHAR), 
	"MODVERSION" NUMBER(12,0) NOT NULL ENABLE, 
	"GENDAT" DATE NOT NULL ENABLE, 
	"GENINFO" VARCHAR2(1000 CHAR) NOT NULL ENABLE, 
	"MODDAT" DATE NOT NULL ENABLE, 
	"MODINFO" VARCHAR2(1000 CHAR) NOT NULL ENABLE, 
	 CONSTRAINT "PK_GIWSDURATIONHISTORYL" PRIMARY KEY ("GIWSDURATIONHISTORYID")
  USING INDEX (CREATE UNIQUE INDEX "PK_GIWSDURATIONHISTORY" ON "GIWSDURATIONHISTORY" ("GIWSDURATIONHISTORYID") 
    )  ENABLE 
   );
```
* Full command including arguments how you run SchemaSpy:
`../../bin/jdk1.8.0_144/bin/java -jar ../../bin/schemaspy/schemaspy-6.1.0-SNAPSHOT_20190409.jar -dp ../../bin/treiber_oracle/ojdbc7_ora12102.jar -cat % -t orathin-service -hostOptionalPort hostnameofmyoracledatabase -db myservicename -s myschemaname  -u myuser -p mypass -o ../../schemas/myschema -norows -lq -imageformat svg `

## Your Environment
* Version used: SchemaSpy@latest (6.1.0 SNAPSHOT)
* Java version: 1.8.0_144

**Comment by npetzall** [comment]
Yeah, I'm the cause of that, I'll look at it. It was supposed to be support for sqlite pk's

**Comment by npetzall** [comment]
Had a look at the code, I think I'll switch the logging to debug. But I'll use your example to verify what I'm thinking.

We are using a JDBC Meta data call in Java that's named getPrimaryKeys, it should return PR_NAME an that's what we are using.

It might be that the JDBC driver is returning two rows for the PrimaryKey:

`PK_GIWSDURATIONHISTORYL` and `PK_GIWSDURATIONHISTORY`

For `PK_GIWSDURATIONHISTORYL` we don't find any index and we log it as warn, for `PK_GIWSDURATIONHISTORY` we find it and mark that index as PK index.

If that is the case, I'm gonna change the logging to debug instead of warn.

The case with sqlite is that it return PK but it has no index for it, and when I changed this change some logic and added the logging. Previously it just ignored it without logging. now it still ignores it, but it's logging it.

**Comment by npetzall** [comment]
I finally took some time to look at this.
getPrimaryKeys returns only one row like this:
```
TABLE_CAT: null
TABLE_SCHEM: DBUSER
TABLE_NAME: GIWSDURATIONHISTORY
COLUMN_NAME: GIWSDURATIONHISTORYID
KEY_SEQ: 1
PK_NAME: PK_GIWSDURATIONHISTORYL
```

If i remove the using index clause, an Index will be created with the name `PK_GIWSDURATIONHISTORYL` it will/should then be correct.

As for the HTML, the key for the index is blue which only means it's an unique index, not actually considered a primary key index by schemaspy.

So I'm a bit unsure if we make a faulty assumption in SchemaSpy that PK_NAME is an index name when it can be either an index or a constraint name. (might differ between dbms).

To get this working correctly we might need to use customSQL (add support for it) JDBC doesn't contain any code to fetch constraints, and I'm using DBBeaver and it has the same problem. It can show constraints, but not that the PK constraint uses a index by a different name.

I assume that the HTML looked the same with the 6.0 and 6.1.0-SNAPSHOT. Index is show with a blue key?

Possible custom sql would be `SELECT null AS table_cat, a.owner AS table_schem, a.table_name, c.column_name AS column_name, c.column_position AS key_seq, a.index_name AS pk_name  FROM all_constraints a LEFT JOIN all_ind_columns c ON c.index_owner = a.owner AND c.table_name = a.table_name  WHERE a.constraint_type = 'P' AND a.owner = :owner AND a.table_name = :table ORDER BY c.column_name`

**Comment by npetzall** [comment]
I added this to ora.properties databaseType
```
selectPrimaryKeysSql=SELECT null AS table_cat, a.owner AS table_schem, a.table_name, c.column_name AS column_name, c.column_position AS key_seq, a.index_name AS pk_name FROM all_constraints a LEFT JOIN all_ind_columns c ON c.index_owner = a.index_owner AND c.table_name = a.table_name WHERE a.constraint_type = 'P' AND a.owner = :owner AND a.table_name = :table ORDER BY c.column_name
```

And added handling for it and got the following
![image](https://user-images.githubusercontent.com/4175123/56836291-16893b00-6877-11e9-939f-ab4049e0e39d.png)
